### PR TITLE
Selenium Observability blog

### PR DESCRIPTION
**Pull request by pujagani**
**Thanks for contributing to the Selenium site and documentation!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://selenium.dev/documentation/en/contributing/) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.

### Description

A blog post on Selenium 4's observability feature and resource

### Motivation and Context

Selenium 4 introduces observability in form of tracing and event logs. The blog is about discussing the need for observability and how to leverage it in Selenium.

### Types of changes

- [ ] Change to the site (I am attaching a screenshot showing the before and after)
- [ ] Code example added (and I also added the example to all translated languages)
- [ ] Improved translation
- [ ] Added new translation (and I also added a notice to each document missing translation)

### Checklist

- [X] I have read the [**contributing**](https://selenium.dev/documentation/en/contributing/) document.
- [X] I have used [hugo](https://gohugo.io/) to render the site/docs locally and I am sure it works.

### CODE REVIEW

File: site_source_files/content/blog/2021/selenium-4-observability.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+Selenium server is now power-packed with [Opentelemetry](https://opentelemetry.io/) to help troubleshoot issues and optimize performance. Thus, making the Selenium server observable! 
+
+### Need for Observability 
+
+Selenium server enables distributed testing. Instead of running the test locally, the tests run on a server that makes “**smart**” decisions to run them on different remote servers. Every such remote server is capable of hosting different types of browsers and browser versions. 
+
+Selenium 3 uses Hub as the coordinator. It receives new session requests and directs them to the appropriate Node. The Node is the remote end that has browser and operating system capabilities. 
+
+Selenium 4 extends this to provide users with a way to set up a full-blown distributed system. Essentially, [Selenium 4](https://www.selenium.dev/documentation/en/grid/grid_4/components_of_a_grid/) splits the Hub into different components with additional enhancements. But the Node’s role remained the same. To ensure the backward compatibility and simplicity of setting up Grid, standalone and hub-node mode are also available in the new Selenium server.
```
Comment (AutomatedTester) [inline site_source_files/content/blog/2021/selenium-4-observability.md]:
```suggestion
Selenium 4 extends this to provide users with a way to set up a full-blown distributed system. Essentially, [Selenium 4](https://www.selenium.dev/documentation/en/grid/grid_4/components_of_a_grid/) splits the Hub into different components with additional enhancements. However, the Node’s role remained the same. To ensure the backward compatibility and simplicity of setting up Grid, standalone and hub-node mode are also available in the new Selenium server. 
```

File: site_source_files/content/blog/2021/selenium-4-observability.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+Description = "Observability in Selenium 4"
+Title = "Observability in Selenium 4"
+Date = 2021-04-26
+Author = "Puja Jagani"
+AuthorLink = "https://twitter.com/pujagani"
+tags = ["selenium", "grid"]
+categories = ["general"]
++++
+
+Selenium server is now power-packed with [Opentelemetry](https://opentelemetry.io/) to help troubleshoot issues and optimize performance. Thus, making the Selenium server observable!
```
Comment (shs96c) [inline site_source_files/content/blog/2021/selenium-4-observability.md]: Starting with "why should I care?" is a nice way to hook people into an article. Perhaps something like...

```
Diagnosing problems when the Selenium server isn't working has never been easy. With Selenium 4, we've added support for [OpenTelemetry](https://opentelemetry.io/) to help you troubleshoot issues and optimize performance. By making the Selenium server observable, we're putting more power into your hands.
```

… or something similar?

File: site_source_files/content/blog/2021/selenium-4-observability.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+AuthorLink = "https://twitter.com/pujagani"
+tags = ["selenium", "grid"]
+categories = ["general"]
++++
+
+Selenium server is now power-packed with [Opentelemetry](https://opentelemetry.io/) to help troubleshoot issues and optimize performance. Thus, making the Selenium server observable! 
+
+### Need for Observability 
+
+Selenium server enables distributed testing. Instead of running the test locally, the tests run on a server that makes “**smart**” decisions to run them on different remote servers. Every such remote server is capable of hosting different types of browsers and browser versions.
```
Comment (shs96c) [inline site_source_files/content/blog/2021/selenium-4-observability.md]: Nit: the tests still run locally. The browsers run remotely. I'm not sure how much it matters, given that I suspect people seldom differentiate the two.

File: site_source_files/content/blog/2021/selenium-4-observability.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+categories = ["general"]
++++
+
+Selenium server is now power-packed with [Opentelemetry](https://opentelemetry.io/) to help troubleshoot issues and optimize performance. Thus, making the Selenium server observable! 
+
+### Need for Observability 
+
+Selenium server enables distributed testing. Instead of running the test locally, the tests run on a server that makes “**smart**” decisions to run them on different remote servers. Every such remote server is capable of hosting different types of browsers and browser versions. 
+
+Selenium 3 uses Hub as the coordinator. It receives new session requests and directs them to the appropriate Node. The Node is the remote end that has browser and operating system capabilities.
```
Comment (shs96c) [inline site_source_files/content/blog/2021/selenium-4-observability.md]: "The Node is the remote end where the browser itself runs"

File: site_source_files/content/blog/2021/selenium-4-observability.md (hunk 1 above)
Comment (shs96c) [reply site_source_files/content/blog/2021/selenium-4-observability.md]: Nit: keep the tense the same throughout the paragraph. Either "split" and "remained" or "splits" and "remains"

File: site_source_files/content/blog/2021/selenium-4-observability.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+Selenium 3 uses Hub as the coordinator. It receives new session requests and directs them to the appropriate Node. The Node is the remote end that has browser and operating system capabilities. 
+
+Selenium 4 extends this to provide users with a way to set up a full-blown distributed system. Essentially, [Selenium 4](https://www.selenium.dev/documentation/en/grid/grid_4/components_of_a_grid/) splits the Hub into different components with additional enhancements. But the Node’s role remained the same. To ensure the backward compatibility and simplicity of setting up Grid, standalone and hub-node mode are also available in the new Selenium server. 
+
+Now picture running a full-blown distributed Selenium Grid with hundreds of Nodes with different browsers and browser versions. Testing at scale in such an infrastructure will involve large volumes of requests. How can one keep track of these requests? Enter Selenium Observability! 
+
+### How does Selenium provide observability?
```
Comment (shs96c) [inline site_source_files/content/blog/2021/selenium-4-observability.md]: Please "title case" as if this were a main heading: "How Does Selenium Provide Observability"

File: site_source_files/content/blog/2021/selenium-4-observability.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+Now picture running a full-blown distributed Selenium Grid with hundreds of Nodes with different browsers and browser versions. Testing at scale in such an infrastructure will involve large volumes of requests. How can one keep track of these requests? Enter Selenium Observability! 
+
+### How does Selenium provide observability? 
+
+Selenium uses OpenTelemetry to instrument tracing and event logs. Tracing keeps track of a request’s lifecycle. As a request moves through the distributed system, the trace of the request will contain all the information of its crucial operations performed along the way. Such an operation is known as span. A span can record timed logs called events, which ideally encapsulate the current state of the system. These are event logs.
```
Comment (shs96c) [inline site_source_files/content/blog/2021/selenium-4-observability.md]: "Such an operation is known as a span" or "Such operations are known as spans"

File: site_source_files/content/blog/2021/selenium-4-observability.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+* [Hub and Node](https://www.selenium.dev/documentation/en/grid/grid_4/setting_up_your_own_grid/#hub-and-node-mode)
+* [Fully Distributed](https://www.selenium.dev/documentation/en/grid/grid_4/setting_up_your_own_grid/#distributed-mode)
+* [Docker](https://github.com/SeleniumHQ/docker-selenium#readme)
+
+Tracing and event logs are available for all the modes. The simplest way to see traces is in the form of console logs. By default, the Selenium server prints logs at the INFO level and above. To pass a logging level of choice while running a Selenium Grid use the [log-level](https://www.selenium.dev/documentation/en/grid/grid_4/advanced_features/observability/#visualizing-traces) flag. Setting the log-level to FINE will display traces and event logs as console logs.
+
+For a large scale system, consuming traces as logs might not be efficient.
+Visualizing and querying traces will quickly help troubleshoot a request failure easily. [Jaeger](https://www.jaegertracing.io/) seamlessly integrates with OpenTelemetry to provide a rich experience of querying, visualizing and collecting request traces. 
+
+Run ``` java -jar selenium-server-4.0.0-<selenium-version>.jar info tracing```.
```
Comment (shs96c) [inline site_source_files/content/blog/2021/selenium-4-observability.md]: `-4.0.0` should be removed from the jar path :)

File: site_source_files/content/blog/2021/selenium-4-observability.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+Tracing and event logs are available for all the modes. The simplest way to see traces is in the form of console logs. By default, the Selenium server prints logs at the INFO level and above. To pass a logging level of choice while running a Selenium Grid use the [log-level](https://www.selenium.dev/documentation/en/grid/grid_4/advanced_features/observability/#visualizing-traces) flag. Setting the log-level to FINE will display traces and event logs as console logs.
+
+For a large scale system, consuming traces as logs might not be efficient.
+Visualizing and querying traces will quickly help troubleshoot a request failure easily. [Jaeger](https://www.jaegertracing.io/) seamlessly integrates with OpenTelemetry to provide a rich experience of querying, visualizing and collecting request traces. 
+
+Run ``` java -jar selenium-server-4.0.0-<selenium-version>.jar info tracing```. 
+
+It provides detailed and updated instructions to set up Jaeger with the Selenium server. 
+Visualizing the traces makes it easy to interpret a trace and understand request flow. Now Grid users can effortlessly trace a request, drill down an error or query them to observe what is happening.
```
Comment (shs96c) [inline site_source_files/content/blog/2021/selenium-4-observability.md]: Nit: "drill down into an error"

File: site_source_files/content/blog/2021/selenium-4-observability.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+It provides detailed and updated instructions to set up Jaeger with the Selenium server. 
+Visualizing the traces makes it easy to interpret a trace and understand request flow. Now Grid users can effortlessly trace a request, drill down an error or query them to observe what is happening. 
+
+{{<figure src="/images/blog/2021/trace.png" alt="Trace example">}}
+
+
+Refer to [Selenium Observability](https://www.selenium.dev/documentation/en/grid/grid_4/advanced_features/observability/) for details. 
+
+### Full-stack tracing
```
Comment (shs96c) [inline site_source_files/content/blog/2021/selenium-4-observability.md]: Full-Stack Tracing

File: site_source_files/content/blog/2021/selenium-4-observability.md [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+Java client binding supports tracing. Full-stack tracing allows tracing a request from the client to the server and back. Though the client is a single component, the client trace will contain spans that help build the request. It is easier to locate a client-side problem and fix the test if needed.
```
Comment (shs96c) [inline site_source_files/content/blog/2021/selenium-4-observability.md]: The Java client binding...

File: site_source_files/content/blog/2021/selenium-4-observability.md [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+
+{{<figure src="/images/blog/2021/trace.png" alt="Trace example">}}
+
+
+Refer to [Selenium Observability](https://www.selenium.dev/documentation/en/grid/grid_4/advanced_features/observability/) for details. 
+
+### Full-stack tracing
+Java client binding supports tracing. Full-stack tracing allows tracing a request from the client to the server and back. Though the client is a single component, the client trace will contain spans that help build the request. It is easier to locate a client-side problem and fix the test if needed. 
+
+We already saw how the server supports tracing and how to leverage it. For the client-side, add the Opentelemetry dependency to your project setup and add the necessary system properties to export the traces to Jaeger. Refer [RemoteWebdriver client](https://www.selenium.dev/documentation/en/remote_webdriver/remote_webdriver_client/#tracing-client-requests) for detailed instructions.
```
Comment (shs96c) [inline site_source_files/content/blog/2021/selenium-4-observability.md]: Nit: "Refer to"

File: site_source_files/content/blog/2021/selenium-4-observability.md [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+
+
+Refer to [Selenium Observability](https://www.selenium.dev/documentation/en/grid/grid_4/advanced_features/observability/) for details. 
+
+### Full-stack tracing
+Java client binding supports tracing. Full-stack tracing allows tracing a request from the client to the server and back. Though the client is a single component, the client trace will contain spans that help build the request. It is easier to locate a client-side problem and fix the test if needed. 
+
+We already saw how the server supports tracing and how to leverage it. For the client-side, add the Opentelemetry dependency to your project setup and add the necessary system properties to export the traces to Jaeger. Refer [RemoteWebdriver client](https://www.selenium.dev/documentation/en/remote_webdriver/remote_webdriver_client/#tracing-client-requests) for detailed instructions.  
+
+### Leveraging traces and event-logs
```
Comment (shs96c) [inline site_source_files/content/blog/2021/selenium-4-observability.md]: Leveraging Traces and Event Logs

**Review by AutomatedTester — APPROVED** [review]

**Review by shs96c — APPROVED** [review]
Thank you for writing this up, @pujagani. LGTM. Ship it! :)

**Review by diemol — APPROVED** [review]
Thank you, @pujagani!

**Pull request merged**